Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -0,0 +1,892 @@
+    h.dist_integralSum_integral_le_of_memBaseSet hpos₂
+      (h₂.unionComplToSubordinate (fun _ _ => min_le_right _ _) hπU hπc₂)
+      (isPartition_unionComplToSubordinate _ _ _ _)
+  simpa [unionComplToSubordinate] using (dist_triangle_right _ _ _).trans (add_le_add H₁ H₂)
+#align box_integral.integrable.dist_integral_sum_le_of_mem_base_set BoxIntegral.Integrable.dist_integralSum_le_of_memBaseSet
+
+/-- If `f` is integrable on `I` along `l`, then for two sufficiently fine tagged prepartitions
+(in the sense of the filter `BoxIntegral.IntegrationParams.toFilter l I`) such that they cover
+the same part of `I`, the integral sums of `f` over `π₁` and `π₂` are very close to each other.  -/
+theorem tendsto_integralSum_toFilter_prod_self_inf_iUnion_eq_uniformity (h : Integrable I l f vol) :
+    Tendsto
+      (fun π : TaggedPrepartition I × TaggedPrepartition I =>
+        (integralSum f vol π.1, integralSum f vol π.2))
+      ((l.toFilter I ×ˢ l.toFilter I) ⊓ 𝓟 {π | π.1.iUnion = π.2.iUnion}) (𝓤 F) := by
+  refine (((l.hasBasis_toFilter I).prod_self.inf_principal _).tendsto_iff
+    uniformity_basis_dist_le).2 fun ε ε0 => ?_
+  replace ε0 := half_pos ε0
+  use h.convergenceR (ε / 2), h.convergenceR_cond (ε / 2); rintro ⟨π₁, π₂⟩ ⟨⟨h₁, h₂⟩, hU⟩
+  rw [← add_halves ε]
+  exact h.dist_integralSum_le_of_memBaseSet ε0 ε0 h₁.choose_spec h₂.choose_spec hU
+#align box_integral.integrable.tendsto_integral_sum_to_filter_prod_self_inf_Union_eq_uniformity BoxIntegral.Integrable.tendsto_integralSum_toFilter_prod_self_inf_iUnion_eq_uniformity
+
+/-- If `f` is integrable on a box `I` along `l`, then for any fixed subset `s` of `I` that can be
+represented as a finite union of boxes, the integral sums of `f` over tagged prepartitions that
+cover exactly `s` form a Cauchy “sequence” along `l`. -/
+theorem cauchy_map_integralSum_toFilteriUnion (h : Integrable I l f vol) (π₀ : Prepartition I) :
+    Cauchy ((l.toFilteriUnion I π₀).map (integralSum f vol)) := by
+  refine' ⟨inferInstance, _⟩
+  rw [prod_map_map_eq, ← toFilter_inf_iUnion_eq, ← prod_inf_prod, prod_principal_principal]
+  exact h.tendsto_integralSum_toFilter_prod_self_inf_iUnion_eq_uniformity.mono_left
+    (inf_le_inf_left _ <| principal_mono.2 fun π h => h.1.trans h.2.symm)
+#align box_integral.integrable.cauchy_map_integral_sum_to_filter_Union BoxIntegral.Integrable.cauchy_map_integralSum_toFilteriUnion
+
+variable [CompleteSpace F]
+
+theorem to_subbox_aux (h : Integrable I l f vol) (hJ : J ≤ I) :
+    ∃ y : F, HasIntegral J l f vol y ∧
+      Tendsto (integralSum f vol) (l.toFilteriUnion I (Prepartition.single I J hJ)) (𝓝 y) := by
+  refine (cauchy_map_iff_exists_tendsto.1
+    (h.cauchy_map_integralSum_toFilteriUnion (.single I J hJ))).imp fun y hy ↦ ⟨?_, hy⟩
+  convert hy.comp (l.tendsto_embedBox_toFilteriUnion_top hJ)
+#align box_integral.integrable.to_subbox_aux BoxIntegral.Integrable.to_subbox_aux
+
+-- faster than `exact` here
```

Comment:
Is this comment stray? Perhaps it is referring to the `refine` in `to_subbox_aux`.